uamqp/message.py

```diff
@@ -109,10 +110,81 @@ def __init__(self,
                 self._body.set(body)
             if msg_format:
                 self._message.message_format = msg_format
-            self.properties = properties
-            self.application_properties = application_properties
-            self.annotations = annotations
-            self.header = header
+            self._properties = properties
+            self._application_properties = application_properties
+            self._annotations = annotations
+            self._header = header
+
+    @property
+    def properties(self):
+        if self._need_further_parse:
+            self._parse_message_properties()
+        return self._properties
+
+    @properties.setter
+    def properties(self, value):
+        if not isinstance(value, MessageProperties):
+            raise TypeError("Properties must be a MessageProperties.")
+        self._message.properties = value.get_properties_obj()
+        self._properties = value
+
+    @property
+    def header(self):
+        if self._need_further_parse:
+            self._parse_message_properties()
+        return self._header
+
+    @header.setter
+    def header(self, value):
+        if not isinstance(value, MessageHeader):
+            raise TypeError("Header must be a MessageHeader.")
+        self._message.header = value.get_header_obj()
+        self._header = value
+
+    @property
+    def footer(self):
+        if self._need_further_parse:
+            self._parse_message_properties()
+        return self._footer
+
+    @property
+    def application_properties(self):
+        if self._need_further_parse:
+            self._parse_message_properties()
+        return self._application_properties
+
+    @application_properties.setter
+    def application_properties(self, value):
+        if not isinstance(value, dict):
+            raise TypeError("Application properties must be a dictionary.")
+        amqp_props = utils.data_factory(value, encoding=self._encoding)
+        self._message.application_properties = amqp_props
+        self._application_properties = value
+
+    @property
+    def annotations(self):
+        if self._need_further_parse:
+            self._parse_message_properties()
+        return self._annotations
+
+    @annotations.setter
+    def annotations(self, value):
+        if not isinstance(value, dict):
+            raise TypeError("Message annotations must be a dictionary.")
+        ann_props = c_uamqp.create_message_annotations(
+            utils.data_factory(value, encoding=self._encoding))
+        self._message.message_annotations = ann_props
```

Comment:
We don't need to do this here because it's done in `def _populate_message_attributes` which is called during send/encode

Reply:
thanks for point out this, removed